src/adapter-types/src/timestamp_selection.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use serde::{Deserialize, Serialize};
+
+/// Whether to use the constraint-based timestamp selection.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum ConstraintBasedTimestampSelection {
+    Enabled,
+    Disabled,
+    Verify,
+}
+
+impl std::default::Default for ConstraintBasedTimestampSelection {
```

Comment:
```suggestion
impl Default for ConstraintBasedTimestampSelection {
```

Or, use [derivative](https://mcarton.github.io/rust-derivative/latest/index.html).